hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java

```diff
@@ -173,41 +173,44 @@ private void isDiscoveryConfigurationConsistent(boolean addressListProvided, boo
                                                     boolean gcpDiscoveryEnabled, boolean azureDiscoveryEnabled,
                                                     boolean kubernetesDiscoveryEnabled, boolean eurekaDiscoveryEnabled,
                                                     boolean discoverySpiEnabled, boolean hazelcastCloudEnabled) {
+        List<String> enabledDiscoveries = new ArrayList<>();
         int count = 0;
         if (addressListProvided) {
             count++;
+            enabledDiscoveries.add("cluster members given explicitly");
         }
         if (awsDiscoveryEnabled) {
             count++;
+            enabledDiscoveries.add("AWS discovery");
         }
         if (gcpDiscoveryEnabled) {
             count++;
+            enabledDiscoveries.add("GCP discovery");
         }
         if (azureDiscoveryEnabled) {
             count++;
+            enabledDiscoveries.add("Azure discovery");
         }
         if (kubernetesDiscoveryEnabled) {
             count++;
+            enabledDiscoveries.add("Kubernetes discovery");
         }
         if (eurekaDiscoveryEnabled) {
             count++;
+            enabledDiscoveries.add("Eureka discovery");
         }
         if (discoverySpiEnabled) {
             count++;
+            enabledDiscoveries.add("Discovery SPI");
         }
         if (hazelcastCloudEnabled) {
             count++;
+            enabledDiscoveries.add("Hazelcast Cloud");
         }
         if (count > 1) {
             throw new IllegalStateException("Only one discovery method can be enabled at a time. "
-                    + "cluster members given explicitly : " + addressListProvided
-                    + ", aws discovery: " + awsDiscoveryEnabled
-                    + ", gcp discovery: " + gcpDiscoveryEnabled
-                    + ", azure discovery: " + azureDiscoveryEnabled
-                    + ", kubernetes discovery: " + kubernetesDiscoveryEnabled
-                    + ", eureka discovery: " + eurekaDiscoveryEnabled
-                    + ", discovery spi enabled : " + discoverySpiEnabled
-                    + ", hazelcast.cloud enabled : " + hazelcastCloudEnabled);
+                    + "Disable one of the following discovery methods by removing it from the configuration, "
```

Comment:
I think the wording is not correct here. It should be rather something like _"Keep only one of following enabled..."_ or _"Disable all but one ..."_

Reply:
Addressed.